## Exercise 1

### Print the number of seconds it would take you to fold 1000 pizza boxes if you were able to maintain the world record pace of 14 pizza boxes in one minute indefinitely.
* [link here](https://www.guinnessworldrecords.com/world-records/103587-most-pizza-boxes-folded-in-one-minute)

**Bonus**: see if you can find a way to print the decimal only to the tenth's place.

In [5]:
## write code here
world_record_pace = 14/60 # [boxes/ min]
n_seconds_1000 = 1000 / world_record_pace
print(f'It would take approximately {n_seconds_1000:.1f} to fold 1000 pizza boxes at world record pace.')


It would take approximately 4285.7 to fold 1000 pizza boxes at world record pace.


Correct output: 4285.714285714286 (4285.7 for bonus)

## Exercise 2

### Imagine you are have a computer program that remembers each word you type for one minute and outputs a string of these typed words seperated by commas. Given the example string below, programmatically find word closest to the end of the alphabet that you typed and print it.

In [10]:
words = 'banana,peanut,xylophone,aardvark,foot,python'
# insert code here

def rev_alphabet(words):  
    l = words.split(",")
    l.sort()
    return l[-1]

print(rev_alphabet(words))

xylophone


Correct output: xylophone

## Exercise 3 (Challenge Problem)

### Now imagine that your computer program from Exercise 2 outputs the words you have typed as a list with each element being one word, but there is an inexplicable number of spaces at the end of each string. While you may be able to fix the program putting in these spaces in the first place, you challenge yourself programmatically determine the shortest word in the list and print it despite these spaces.

**Note**: This problem can be solved with methods not yet discussed in this course, but I urge you to limit yourself to methods/functions we have already discussed.

**Hint**: The *sorted* function has an optional argument *key* that you can pass a function to such that *sorted* uses that function to determine the order of sorting (see documentation [here](https://www.w3schools.com/python/ref_func_sorted.asp))

In [30]:
words_lst = ['banana ', ' peanut  ', 'xylophone  ', 'aardvark  ', 'foot       ', 'python    ']
# insert code here
import numpy as np
def shortest_word(words_lst):  
    l = np.array([word.strip() for word in words_lst])
    word_sizes = np.array([len(word) for word in l])
    shortest_length = np.min(word_sizes)
    return l[word_sizes==shortest_length][0]
print(shortest_word(words_lst))

foot


Correct output: foot
